File: docs/src/test/java/sample/javadsl/FetchMetadata.java
package sample.javadsl;

// #metadata
        import akka.actor.ActorRef;
        import akka.kafka.KafkaConsumerActor;
        import akka.kafka.Metadata;
        import akka.pattern.PatternsCS;
        import akka.util.Timeout;
        import org.apache.kafka.common.TopicPartition;

        import java.util.concurrent.CompletionStage;
        import java.util.concurrent.TimeUnit;

// #metadata


public class FetchMetadata extends ConsumerExample {

    public static void main(String[] args) {
        new FetchMetadata().demo();
    }

    void demo() {
        // #metadata
        // Create kafka consumer actor to be used with Consumer.plainExternalSource or committableExternalSource
        ActorRef consumer = system.actorOf((KafkaConsumerActor.props(consumerSettings)));

        // ... create source ...

        CompletionStage<Metadata.Topics> topicsStage = PatternsCS.ask(consumer, new Metadata.GetCommittedOffset(new TopicPartition("f", 0)), new Timeout(2, TimeUnit.SECONDS))
                .thenApply(reply -> ((Metadata.Topics) reply));

        // print response
        topicsStage
                .thenApply(Metadata.Topics::getResponse)
                .thenAccept(responseOption -> responseOption.ifPresent(map -> map.forEach((topic, partitionInfo) ->
                        partitionInfo.forEach(info ->
                                System.out.println(topic + ": " + info.toString())
                        ))));

        // #metadata
    }
}

File: docs/src/test/scala/sample/scaladsl/FetchMetadata.scala
package sample.scaladsl

// #metadata
import akka.actor.ActorRef
import akka.kafka.KafkaConsumerActor
import akka.kafka.Metadata
import akka.pattern.ask
import akka.util.Timeout

import scala.concurrent.Future
import scala.concurrent.duration._

// #metadata

// Connect a Consumer to Producer, mapping messages one-to-many, and commit in batches
object FetchMetadata extends ConsumerExample {

  def main(args: Array[String]): Unit = {
    // #metadata
    implicit val timeout = Timeout(5.seconds)

    val consumer: ActorRef = system.actorOf(KafkaConsumerActor.props(consumerSettings))

    // ... create source ...

    val topicsFuture: Future[Metadata.Topics] = (consumer ? Metadata.ListTopics).mapTo[Metadata.Topics]

    topicsFuture.map(_.response.foreach { map =>
      map.foreach {
        case (topic, partitionInfo) =>
          partitionInfo.foreach { info =>
            println(s"$topic: $info")
          }
      }
    })
    // #metadata

  }
}
